tools/wptrunner/wptrunner/testrunner.py

```diff
@@ -703,6 +703,13 @@ def test_ended(self, test, results):
                 expected_fail_message = test.expected_fail_message(result.name)
                 if expected_fail_message is not None and result.message.strip() != expected_fail_message:
                     is_unexpected = True
+                    if result.status in known_intermittent:
+                        known_intermittent.remove(result.status)
+                    elif len(known_intermittent) > 0:
```

Comment:
Without noticing `not is_unexpected` earlier, it was a bit difficult to see why this was the case. Maybe a comment before the first `elif` along the lines of "`result.status` should equal `expected` at this point".

Reply:
Sure. I'd like to leave this to a separate PR because the upstream CI is fragile, and am not really sure if the test can still pass with a no-op change.